modules/tinymce/src/core/main/ts/fmt/MatchFormat.ts

```diff
@@ -67,65 +67,65 @@ const matchName = (dom: DOMUtils, node: Node, format) => {
   }
 };
 
-const matchItems = (dom: DOMUtils, node: Node, format, itemName: string, similar: boolean, vars: FormatVars) => {
-  let key, value;
+const matchItems = (dom: DOMUtils, node: Node, format: Format, itemName: string, similar: boolean, vars: FormatVars): boolean => {
   const items = format[itemName];
-  let i;
 
   // Custom match
   if (format.onmatch) {
-    return format.onmatch(node, format, itemName);
+    const onMatch: (node: Node, format: Format, name: string) => boolean = format.onmatch;
+    return onMatch(node, format, itemName);
   }
 
   // Check all items
   if (items) {
     // Non indexed object
     if (typeof items.length === 'undefined') {
-      for (key in items) {
+      for (const key in items) {
         if (Obj.has(items, key)) {
-          if (itemName === 'attributes') {
-            value = dom.getAttrib(node, key);
-          } else {
-            value = FormatUtils.getStyle(dom, node, key);
+          const value = itemName === 'attributes' ? dom.getAttrib(node, key) : FormatUtils.getStyle(dom, node, key);
+          const expectedValue = FormatUtils.replaceVars(items[key], vars);
+
+          if (!value && !expectedValue) {
```

Comment:
This here is the actual bug fix for the formatter change. If the expected value is falsey, and the value itself is falsey, then for this key we have a match (thus move onto the next key).

If this condition needs tweaking at all to account for `similar` or `format.exact` (I don't know much about either value) please let me know.

Reply:
We also need a changelog entry for this fix.

- [ ] Changelog entry

Reply:
Added